src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -393,7 +451,338 @@ public virtual bool TryInsertExpansion(int startPositionInSubjectBuffer, int end
+            });
+
+            if (symbols.OfType<IMethodSymbol>().Any())
+            {
+                XNamespace snippetNamespace = "http://schemas.microsoft.com/VisualStudio/2005/CodeSnippet";
+
+                var methodName = dataBufferSpan.GetText();
+
+                var template = $"{methodName}($placeholder$)$end$";
+
+                var snippet = new XDocument(
+                    new XDeclaration("1.0", "utf-8", null),
+                    new XElement(
+                        snippetNamespace + "CodeSnippets",
+                        new XElement(
+                            snippetNamespace + "CodeSnippet",
+                            new XAttribute(snippetNamespace + "Format", "1.0.0"),
+                            new XElement(
+                                snippetNamespace + "Header",
+                                new XElement(
+                                    snippetNamespace + "SnippetTypes",
+                                    new XElement(snippetNamespace + "SnippetType", new XText("Expansion"))),
+                                new XElement(snippetNamespace + "Title", new XText(methodName)),
+                                new XElement(snippetNamespace + "Author", "Microsoft"),
+                                new XElement(snippetNamespace + "Description"),
+                                new XElement(snippetNamespace + "HelpUrl"),
+                                new XElement(snippetNamespace + "Shortcut", methodName)),
+                            new XElement(
+                                snippetNamespace + "Snippet",
+                                new XElement(
+                                    snippetNamespace + "Declarations",
+                                    new XElement(
+                                        snippetNamespace + "Literal",
+                                        new XAttribute(snippetNamespace + "Editable", "true"),
+                                        new XElement(snippetNamespace + "ID", new XText("placeholder")),
+                                        new XElement(snippetNamespace + "ToolTip", new XText("")),
+                                        new XElement(snippetNamespace + "Default", new XText("")),
+                                        new XElement(snippetNamespace + "Function"))),
+                                new XElement(
+                                    snippetNamespace + "Code",
+                                    new XAttribute(snippetNamespace + "Language", "csharp"),
+                                    new XCData(template))))));
+
+                var doc = new DOMDocumentClass();
+                if (doc.loadXML(snippet.ToString(SaveOptions.OmitDuplicateNamespaces)))
+                {
+                    var methodSymbols = symbols.OfType<IMethodSymbol>().ToImmutableArray();
+                    _symbols = methodSymbols;
+                    _symbol = null;
```

Comment:
plz doc (here or on the fields).  i don't get the relation between these two value.

Reply:
➡️ Tracking this as part of https://github.com/dotnet/roslyn/pull/49916#discussion_r544816693